app/queries/comments/community_wellness_query.rb

```diff
@@ -18,7 +18,27 @@
 module Comments
   class CommunityWellnessQuery
     def self.call
-      ActiveRecord::Base.connection.execute(sql_query).to_a
+      if limit_query_rollout?
+        ActiveRecord::Base.connection.execute(limit_release_date_sql_query).to_a
+      else
+        ActiveRecord::Base.connection.execute(sql_query).to_a
+      end
+    end
+
+    # We are rolling out the badge feature to consider comments posted May 1st,
+    # 2022 or later. The logic here consists in checking if 33 weeks have passed
+    # since May 1st, 2022. The method should return `true` if we need to
+    # restrict the comment dates, and it will return `false` if we are good to
+    # carry on without that restriction (i.e. it's been more than 33 weeks since
+    # the rollout).
+    #
+    # We can officially remove this query+logic on December 19, 2022 at which
+    # time we will continue to use the default query.
+    #
+    # Read more about this here:
+    # https://github.com/forem/forem/issues/17310#issuecomment-1118554640
+    def self.limit_query_rollout?
+      Time.zone.parse("2022-05-01") > 33.weeks.ago
     end
```

Comment:
In order to rollout the badge streak we are limiting the SQL query based on this validation. This will allow everyone to "start earning the badge starting from week 1".

I'll be adding an internal reminder (Slack channel reminder) to remove this logic when the time is right. I can even leave a Draft PR (after this one is merged) that can quickly be put "up to date" (merge main) and then merged when that time comes, to make that process much easier for me or anyone that will handle that.

I left this commented inline to the best of my ability as well, but you can read more about the why here: https://github.com/forem/forem/issues/17310#issuecomment-1118554640